### PR TITLE
Prombench Alerting 

### PR DESCRIPTION
**Pull request by sipian**
Closes #64 
This PR modifies the following things

-  alertmanager deployment & config files.
- Alerting rules are added for `CPU` & `memory`.
- Namespaces are changed from `prombench-<PR_NUMBER>` to `<PR_NUNMBER>`  
- Prometheus-meta service-discovers alertmanager pods(instead of static_configs)

A sample deployment can be found [here](http://35.237.152.93/prometheus-meta/alerts) 
A sample comment posted by alertmanager-github-receiver can be found [here](https://github.com/sipian/prometheus/pull/12#issuecomment-410467512)

### CODE REVIEW

**Comment by krasi-georgiev** [comment]
this close https://github.com/prometheus/prombench/issues/32 right? not 64

If we can change rules from the UI I would say -  lets leave it open at the begining so we can quickly change rules, but will close later or think about some authentication.

**Comment by sipian** [comment]
This closes #64 because this PR also deploys `alertmanager-github-receiver`.
Rules cannot be updated from alert-manager UI because rules are added in prometheus config.
alert-manager UI is just to see status of fired alerts & silence them if needed

**Comment by krasi-georgiev** [comment]
Is there anything else that would be useful if we have the GUI exposed?

**Comment by sipian** [comment]
I don't think so. 
Maybe if maintainer wants to see a complete list of fired alerts?

**Comment by sipian** [comment]
@krasi-georgiev 
Can you merge this so that I can push prow-node.json grafana-dashboard in #87

**Comment by krasi-georgiev** [comment]
did you test it in your test prow installation how well it works?

**Comment by sipian** [comment]
I tested on prombench cluster. Alerts are being fired and appropriate comments are being written in the PR. I reduced the alerting threshold to a small value to trigger alerts. 

I haven't tested the actual thresholds though

File: components/prombench/manifests/results/alerting_rules.yaml [hunk 1]
```diff
@@ -10,25 +10,25 @@ data:
...
     - name: cpu
       rules:
       - alert: NodeCPUUsage
-        expr: (100 - (avg by(instance, node, namespace) (rate(node_cpu_seconds_total{job="node-exporter",mode="idle",namespace=~"[[:digit:]]+",node=~".+"}[5m]))
-          * 100)) > 90
+        expr: ((rate(process_cpu_seconds_total{job="prometheus",namespace=~"prombench-[[:digit:]]+",prometheus=~"test-pr-.+"}[5m]) 
+          - on(namespace) rate(process_cpu_seconds_total{job="prometheus",namespace=~"prombench-[[:digit:]]+",prometheus=~"test-[^pr-].+"}[5m]) ) 
+          / on (namespace) rate(process_cpu_seconds_total{job="prometheus",namespace=~"prombench-[[:digit:]]+",prometheus=~"test-[^pr-].+"}[5m])*100)
+          > 15
         for: 2m
```
Comment (sipian) [inline components/prombench/manifests/results/alerting_rules.yaml]: Is this value sufficient?

**Comment by sipian** [comment]
@krasi-georgiev PTAL

File: components/prombench/manifests/results/alertmanager_config.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    global:
+      resolve_timeout: 5m
+
+    # The root route on which each incoming alert enters.
+    route:
+      group_by: ['namespace']
+      group_wait: 30s
+      group_interval: 5m
+      repeat_interval: 4h 
+
```
Comment (sipian) [inline components/prombench/manifests/results/alertmanager_config.yaml]: These values are the default values in alertmanager.

File: components/prombench/manifests/results/alertmanager_config.yaml [hunk 3]
```diff
@@ -0,0 +1,24 @@
+kind: ConfigMap
+apiVersion: v1
```
[the remark this answers is not in the document]
Comment (sipian) [reply components/prombench/manifests/results/alertmanager_config.yaml]: yeah will do that

File: components/prombench/manifests/results/alertmanager_deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    name: alertmanager
+  name: alertmanager
+spec:
+  selector:
+    app: alertmanager
+  type: NodePort
```
[the remark this answers is not in the document]
Comment (sipian) [reply components/prombench/manifests/results/alertmanager_deployment.yaml]: I had added `NodePort` when I was exposing it on nginx. I will update it to clusterIP.

File: components/prombench/manifests/results/alertmanager_config.yaml (hunk 2 above)
Comment (krasi-georgiev) [reply components/prombench/manifests/results/alertmanager_config.yaml]: and how does this work with the `for: 2m` inside the rules ?

**Comment by sipian** [comment]
image
alertmanager-github-receiver provides the link to the prometheus expression which fired the alert.
Problem is sometimes the link does not work :stuck_out_tongue: (some error like missing paranthesis,etc)

File: components/prombench/manifests/results/alertmanager-github-receiver.yaml [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    matchLabels:
+      app: alertmanager-github-receiver
+  template:
+    metadata:
+      labels:
+        app: alertmanager-github-receiver
+    spec: 
+      containers:
+      - name: alertmanager-github-receiver
+        image: docker.io/krasimir/alertmanager-github-receiver
```
Comment (geekodour) [inline components/prombench/manifests/results/alertmanager-github-receiver.yaml:17]: where can i find more about this docker image? is this the modified version of https://github.com/m-lab/alertmanager-github-receiver ? I am unable to understand how it figures out the issue to comment on.

link to docker hub: https://hub.docker.com/r/krasimir/alertmanager-github-receiver

cc @krasi-georgiev @sipian

Comment (krasi-georgiev) [reply components/prombench/manifests/results/alertmanager-github-receiver.yaml:17]: I had a quick look and can't figure it out either.
It should be just few lines to add this functionality and than can open a pr against `m-lab/alertmanager-github-receiver` to add it upstream.

**Comment by krasi-georgiev** [comment]
superseded by https://github.com/prometheus/prombench/pull/208

**Pull request closed without merging**